.TH std::exception::exception 3 "2018.03.28" "http://cppreference.com" "C++ Standard Libary"
.SH NAME
std::exception::exception \- std::exception::exception

.SH Synopsis
   exception();                         \fB(1)\fP
   exception( const exception& other ); \fB(2)\fP

   Constructs new exception object.

   1) Default constructor. what() returns empty string.
   2) Copy constructor. Initializes the contents with those of other

.SH Parameters

   other - another exception to assign the contents of

.SH Exceptions

   \fI(none)\fP                    \fI(until C++11)\fP
   noexcept specification:  
   noexcept                  \fI(since C++11)\fP
     

.SH Notes

   Because copying std::exception is not permitted to throw exceptions, when derived
   classes (such as std::runtime_error) have to manage a user-defined diagnostic
   message, it is typically implemented as a copy-on-write string.

.SH Category:

     * unconditionally noexcept

   Hidden categories:

     * Pages with unreviewed unconditional noexcept template
     * Pages with unreviewed noexcept template
